src/python_testing/TC_TSTAT_2_2.py

```diff
@@ -592,15 +592,16 @@ async def test_TC_TSTAT_2_2(self):
         # Test Harness Reads MinSetpointDeadBand attribute from Server DUT and verifies that the value is within range
         if self.pics_guard(hasAutoModeFeature):
             val = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.MinSetpointDeadBand)
-            asserts.assert_less_equal(val, 25)
+            asserts.assert_less_equal(val, 127)
+            asserts.assert_greater_equal(val, 0)
 
             if self.pics_guard(self.check_pics("TSTAT.S.M.MinSetpointDeadBandWritable")):
                 # Test Harness Writes a value back that is different but valid for MinSetpointDeadBand attribute
-                await self.write_single_attribute(attribute_value=cluster.Attributes.MinSetpointDeadBand(5), endpoint_id=endpoint)
+                await self.write_single_attribute(attribute_value=cluster.Attributes.MinSetpointDeadBand(20), endpoint_id=endpoint)
 
-                # Test Harness Reads it back again to confirm the successful write of MinSetpointDeadBand attribute
-                val = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.MinSetpointDeadBand)
-                asserts.assert_equal(val, 5)
+                # Test Harness Reads it back again to confirm the write of MinSetpointDeadBand attribute is discarded
```

Comment:
we should check that 20 is not the same as val. What if the value is exactly 20 already?